### PR TITLE
View switching button

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards implementing view switching

- [x] implemented button to switch between ride flow and schedule and back
- [x] changed widget tree around for empty page to display properly for both regular home page and schedule view

### Test Plan 

Confirmed that schedule button works for both pages in the ride flow and empty rides widget shows up appropriately
https://user-images.githubusercontent.com/60579411/114124907-6c6c1780-98c3-11eb-81fa-a4047fbb7d54.mp4

### CODE REVIEW

File: lib/pages/Rides.dart [hunk 1]
```diff
@@ -211,59 +259,46 @@ class _RidesState extends State<Rides> {
...
-          onDropoff: () {
-            setState(() {
-              selectedRideIDs.forEach((String id) => finishRide(context, ridesProvider.currentRides.where((ride) => ride.id == id).single));
-              selectedRideIDs = [];
-            });
-          },
-          selectCallback: _selectRide,
-        ),
+    Widget page = SafeArea(
+      child: LoadingOverlay(
```
Comment (tyang98) [inline lib/pages/Rides.dart:269]: This overlay works effectively, nice work!

File: lib/pages/Rides.dart [hunk 2]
```diff
@@ -322,7 +358,13 @@ class RideGroup extends StatelessWidget {
               child: RideGroupTitle(title, rides.length),
             );
           }
-          Widget w = RideCard(rides[index]);
+          Widget w = Opacity(
+              opacity: interactive ? 1 : 0.5,
+              child: RideCard(rides[index])
+          );
+          if (!interactive) {
+            w = IgnorePointer(child: w);
```
Comment (tyang98) [inline lib/pages/Rides.dart:366]: What is the purpose of using an IgnorePointer?

**Review by tyang98 — APPROVED** [review]
The implementation of view switching is sound and the transitions are smooth. I think the animation with regards to gradually "removing" rides from the screen when dropping off riders is really nice. Nice work!

File: lib/pages/Rides.dart (hunk 2 above)
Comment (helenxuyang) [reply lib/pages/Rides.dart:366]: The ride cards are usually tappable to start the ride flow, but shouldn't be in the schedule view, so the IgnorePointer makes them not tappable

**Pull request merged**